### ISSUE TITLE
`$lastModifiedDate` updates on save still not really working

### ISSUE THREAD

**Issue by DMXL** [sidmishraw/topper#5]
User configurations:
```json
"topper.headerTemplates": [
    {
      "defaultCStyled": {
        "headerBegin": "/**",
        "headerPrefix": " *",
        "headerEnd": " */",
        "template": [
          "${headerBegin}",
          "${headerPrefix} Copyright 2018 ${copyright} All rights reserved.",
          "${headerPrefix}",
          "${headerPrefix} @Filename: ${fileName}",
          "${headerPrefix} @Description: ${description}",
          "${headerPrefix} @Author: ${author}",
          "${headerPrefix} @Date: ${createdDate}",
          "${headerPrefix} @Last modified by: ${author}",
          "${headerPrefix} @Last modified time: ${lastModifiedDate}",
          "${headerEnd}"
        ]
      }
    }
  ],
  "topper.lastModified": "@Last modified time:",
```
Tried cmd+s and saving from menu. The `${lastModifiedDate}` part didn't seem to get updated at any point. I'm on macOS 10.13.3 and VS Code 1.23.1, if this is helpful.

**Comment by sidmishraw** [comment]
It is a known issue. Right now, the `last-modified-date/time`is matched with a fixed string. I'll be externalizing those props in the next release –– along with the custom copyright and year feature.

Workaround:
- Make use of `@last-modified` for now.

P.S. - If you would like any other features, please feel free to open enhancement issues.

**Comment by joket1999** [comment]
@sidmishraw The same problem with me. BTW, the Topper extension is very usefully for me.

**Comment by sidmishraw** [comment]
@tanzhihang Thanks for using Topper :)

I made this plugin because VS Code didn't have one header template plugin like Sublime's [FileHeader plugin](https://github.com/shiyanhui/FileHeader). However, unlike FileHeader, I don't want Topper to mess up your existing code –– FileHeader had a bug, it used to overwrite the first line of the file ( I hope it has been fixed, haven't checked it myself).

Btw, I was inspired by the plugin [fileheadercomment](https://github.com/doi/fileheadercomment).

**Comment by sidmishraw** [comment]
Topper has a contribution point in the settings.json file named `topper.lastModified`. Its value is a string and by default it is `@last-modified`. By defining it to your preferred last-modified field name, the topper-watcher will start auto-updating the date-time feature.

> Note: I had added this feature long back but had forgotten to document it -- my bad :/

The pull request from @Drakesinger #6 fixes the regex issue and makes the last-modified field value more flexible and has been merged into master. The changes are now available in the new version of topper -- v0.4.0, already pushed to the marketplace.

Example scenario:
If you have defined your last-modified field in your header template as:
```
"${headerPrefix} modified ${lastModifiedDate}",
```
then, you should set the value of `topper.lastModified` to `modified`. This is enough for topper to start watching it and auto-update the last-modified timestamp.

P.S. Please feel free to send in bugs and PRs, and thanks for using topper :)
